### DISCUSSION TITLE
dependsOn issue for multiple fields

### DISCUSSION THREAD

**Discussion by elizdlt202** [laravel/nova-issues#4062] [Q&A] [answered]
- Laravel Version: 8
- Nova Version: 4.2.3
- PHP Version: 8
- Browser type and version: chrome

### Description:

In the latest release I thought the "dependsOn" issue would be fixed.

I have one field -> product -> where I want to select the product_name.

And based on the product selected I have fields that will have different ad size options.
Each of these fields ( the  different ad size select fields) are dependent on the product_name field. 
But all these adsize select options pertain to the same field 'adsize_label'.
This causes an  issue for the "dependsOn"

![Screen Shot 2022-04-20 at 11 03 05 AM](https://user-images.githubusercontent.com/49735067/164294295-ae9ebd6a-a1c3-47d7-93cf-f84074a963d2.png)
![Screen Shot 2022-04-20 at 11 02 57 AM](https://user-images.githubusercontent.com/49735067/164294304-7e5f86fd-438e-4b79-92db-7a5c0de95fb4.png)

**Comment by elizdlt202** [comment]
https://user-images.githubusercontent.com/49735067/164296465-4cf3e57f-1419-4d94-9a07-e94c9679ecb9.mov

**Comment by Nks** [comment]
Similar error with another field https://github.com/laravel/nova-issues/issues/4045

**Comment by elizdlt202** [comment]
@Nks  I think its a bit different. 
But the dependsOn - great new feature - for sure has issues...
Hope you find your answers!

**Comment by crynobone** [comment, marked as answer]
You should declare `adsize_label` only once and use multiple if else to set `show()` and `options()`.

```php
Select::make('AdSize', 'adsize_label')->options([])
  ->dependsOn('product_name', function ($field, $request, $formData) {
    if ($formData->product_name === 'A') { 
      $field->options(/* ... */)->show();
    } elseif ($formData->product_name === 'B') { 
      $field->options(/* ... */)->show();
    } else {
      $field->hide();
    }
  }),
```

**Reply by elizdlt202** [reply]
wow, I didn't know that you can set the options within the dependsOn
I tested your example - works!

Thank you ! Thank you! Saved me.
